backend/gsr_booking/views.py

```diff
@@ -37,83 +49,87 @@ class UserViewSet(viewsets.ReadOnlyModelViewSet):
             )
         )
         return queryset
 
     @action(detail=True, methods=["get"])
-    def invites(self, request, username=None):
+    def invites(self, request: Request, username: Optional[str] = None) -> Response:
         """
         Retrieve all invites for a given user.
         """
 
         user = get_object_or_404(User, username=username)
+        request_user: Any = self.request.user
         return Response(
             GroupMembershipSerializer(
                 GroupMembership.objects.filter(
-                    user=user, accepted=False, group__in=self.request.user.booking_groups.all()
+                    user=user, accepted=False, group__in=request_user.booking_groups.all()
                 ),
                 many=True,
             ).data
         )
 
 
-class GroupMembershipViewSet(viewsets.ModelViewSet):
+class GroupMembershipViewSet(viewsets.ModelViewSet[GroupMembership]):
     filter_backends = [DjangoFilterBackend]
     filterset_fields = ["user", "group"]
     permission_classes = [IsAuthenticated]
     queryset = GroupMembership.objects.all()
     serializer_class = GroupMembershipSerializer
 
-    def get_queryset(self):
-        if not self.request.user.is_authenticated or not hasattr(self.request.user, "memberships"):
+    def get_queryset(self) -> QuerySet[GroupMembership, Manager[GroupMembership]]:
+        user: Any = self.request.user
+        if not user.is_authenticated:
             return GroupMembership.objects.none()
+
         return GroupMembership.objects.filter(
-            Q(id__in=self.request.user.memberships.all())
+            Q(id__in=user.memberships.all())
             | Q(
                 group__in=Group.objects.filter(
-                    memberships__in=GroupMembership.objects.filter(user=self.request.user, type="A")
+                    memberships__in=GroupMembership.objects.filter(user=user, type="A")
                 )
             )
         )
 
     @action(detail=False, methods=["post"])
-    def invite(self, request):
+    def invite(self, request: Request) -> Response | HttpResponseForbidden:
         """
         Invite a user to a group.
         """
         group_id = request.data.get("group")
         group = get_object_or_404(Group, pk=group_id)
 
         # don't invite when user already in group
-        if group.has_member(request.user):
+        if group.has_member(cast(Any, request.user)):
             return HttpResponseForbidden()
 
         return Response({"message": "invite(s) sent."})
 
     @action(detail=True, methods=["post"])
-    def accept(self, request, pk=None):
+    def accept(
+        self, request: Request, pk: Optional[int] = None
+    ) -> Response | HttpResponseForbidden:
         membership = get_object_or_404(GroupMembership, pk=pk, accepted=False)
-        if membership.user is None or membership.user != request.user:
+        user = cast("UserType", request.user)
```

Comment:
These casts (here and below) are inconsistent. It also seems clunky that are forced to use `cast`. Is this common? If not maybe there's a better way to deal with our User type?